augur/tasks/github/pull_requests/core.py

```diff
@@ -1,6 +1,94 @@
 import logging
-
-from typing import Dict, List, Tuple, Optional
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Dict, List, Tuple, Optional, TypedDict, Any, Union, Literal
+
+# custom type definitions for better type safety and documentation
+class ContributorData(TypedDict, total=False):
```

Comment:
These classes seem quite long and potentially duplicated with the ORM model class objects that already exist in the data models.